crates/biome_glob/src/lib.rs

```diff
@@ -1,31 +1,86 @@
-use biome_rowan::{TextRange, TextSize};
+//! biome_glob provides a glob and glob list with exceptions matching.
```

Comment:
It's unclear what "glob list with expectations" means, especially when I look at the source code where only `Glob` is exposed. See if this suggestion can work

```suggestion
//! biome_glob provides a globbing functionality. When listing the globs to match, it also possible to provide globs that function as "expectations" by prefixing the globs with `!`. 
```